src/CommunityToolkit/Xamarin.CommunityToolkit/Views/SegmentedView/SegmentedView.shared.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public class SegmentedView : View
+	{
+		public event EventHandler<SelectedItemChangedEventArgs> SelectedIndexChanged;
+
+		public static BindableProperty ColorProperty = BindableProperty.Create(nameof(Color), typeof(Color), typeof(SegmentedView));
+		public static BindableProperty SelectedTextColorProperty = BindableProperty.Create(nameof(SelectedTextColor), typeof(Color), typeof(SegmentedView));
+		public static BindableProperty NormalTextColorProperty = BindableProperty.Create(nameof(NormalTextColor), typeof(Color), typeof(SegmentedView));
+		public static BindableProperty SelectedIndexProperty = BindableProperty.Create(nameof(SelectedIndex), typeof(int), typeof(SegmentedView), 0, propertyChanged: OnSegmentSelected);
+		public static BindableProperty DisplayModeProperty = BindableProperty.Create(nameof(DisplayMode), typeof(SegmentMode), typeof(SegmentedView));
+
+		public IEnumerable<string> Items { get; } = new LockableObservableListWrapper();
+
+		public static readonly BindableProperty ItemsSourceProperty =
+			BindableProperty.Create(nameof(ItemsSource), typeof(IEnumerable), typeof(SegmentedView), default(IEnumerable),
+									propertyChanged: OnItemsSourceChanged);
+
+		public static readonly BindableProperty SelectedItemProperty =
+			BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(SegmentedView), null, BindingMode.TwoWay,
+									propertyChanged: OnSelectedItemChanged);
+
+		public IEnumerable<object> ItemsSource
+		{
+			get => (IEnumerable<object>)GetValue(ItemsSourceProperty);
+			set => SetValue(ItemsSourceProperty, value);
+		}
+
+		public object SelectedItem
+		{
+			get => GetValue(SelectedItemProperty);
+			set => SetValue(SelectedItemProperty, value);
+		}
+
+		public SegmentMode DisplayMode
+		{
+			get => (SegmentMode)GetValue(DisplayModeProperty);
+			set => SetValue(DisplayModeProperty, value);
+		}
+
+		public static readonly BindableProperty CornerRadiusProperty =
+			BindableProperty.Create(nameof(CornerRadius), typeof(CornerRadius), typeof(SegmentedView), new CornerRadius(6.0));
+
+		public CornerRadius CornerRadius
+		{
+			get => (CornerRadius)GetValue(CornerRadiusProperty);
+			set => SetValue(CornerRadiusProperty, value);
+		}
+
+		BindingBase itemDisplayBinding;
+
+		public BindingBase ItemDisplayBinding
+		{
+			get => itemDisplayBinding;
+
+			set
+			{
+				if (itemDisplayBinding == value)
+					return;
+
+				OnPropertyChanging();
+				itemDisplayBinding = value;
+				ResetItems();
+				OnPropertyChanged();
+			}
+		}
+
+		static readonly BindableProperty DisplayProperty =
+			BindableProperty.Create("Display", typeof(string), typeof(SegmentedView), default(string));
```

Comment:
```suggestion
		/// <summary>
		/// Backing BindableProperty for the <see cref="Display"/> property.
		/// </summary>
		static readonly BindableProperty DisplayProperty =
			BindableProperty.Create("Display", typeof(string), typeof(SegmentedView), default(string));
```

Reply:
This one is tricky. I don't have a property for "Display" since I am planning on doing some work with this in the future. Such as, letting the dev set a DisplayMember when binding. But not sure how to get there just yet.